source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/overview/products/item/tasks.tsx

```diff
@@ -13,70 +13,137 @@ import { useCurrentUser } from '~/data/containers/with-current-user';
+          const taskUserId = remoteIdentityFrom(dataStore, user).keys.remoteId;
+          const currentUserId = remoteIdentityFrom(dataStore, currentUser).keys.remoteId;
 
-  if (tasks.length === 0) {
-    return <div className='w-100 p-sm p-b-md m-l-md fs-13'>{t('tasks.noTasksTitle')}</div>;
+          if (taskUserId === currentUserId) {
+            foundCurrentUser = true;
+            workTask = task;
+          } else if (!workTask) {
+            // If there isn't a return task set, set it to this entry so that an elapsed time
+            // can be calculated, even if there isn't one for the current user.
+            workTask = task;
+          }
+        });
+      }
+      return [foundCurrentUser, workTask];
+    } catch (e) {
+      console.debug(
+        'error occurred',
+        e,
+        'we probably need to PR to orbit.js, as this is a race condition scenario that causes the error',
+        'variables that are in use that may be related to this error: ',
+        ['product: ', product, 'currentUser: ', currentUser]
+      );
+    }
+  };
+  const formatTaskLine = (transition) => {
+    // TODO: activityName will probably need to be translated
+    //       need to sync on names of those, or a strategy of how to translate.
+    const attributes = transition.attributes;
+    let allowedNames = attributes['allowed-user-names'];
+    if (!allowedNames) {
+      allowedNames = t('tasks.scriptoria');
+    }
+    const activityName = attributes['initial-state'];
+    let waitTime = '';
+    // Tasks assigned to the system don't have a user task
+    // They will get their wait time from the current transition record
+    if (task) {
+      const { dateCreated } = attributesFor(task);
+      waitTime = relativeTimeAgo(dateCreated);
+      // NOTE: without this check, we can enter a race condition where a product is removed
+      //       from the store, but the task still exists.
+      const taskProduct = dataStore.cache.query((q) => q.findRelatedRecord(task, 'product'));
+      if (!taskProduct) return 'Task has no product!';
+    } else {
+      const dateTransition = attributes['date-started'];
+      if (dateTransition) {
+        waitTime = relativeTimeAgo(dateTransition);
+      }
+    }
+    if (currentUserTask) {
+      return (
+        <div key={transition.id}>
+          <span className='red-text'>{t('tasks.waiting', { waitTime })}</span>&nbsp;
+          {t('tasks.forNames', { allowedNames, activityName })}
+          <Link className='m-l-md bold uppercase' to={pathToWorkflow(task)}>
+            {t('common.continue')}
+          </Link>
+        </div>
+      );
+    } else {
+      return (
+        <div key={transition.id}>
+          <span className='red-text'>{t('tasks.waiting', { waitTime })}</span>&nbsp;
+          {t('tasks.forNames', { allowedNames, activityName })}
+        </div>
+      );
+    }
+  };
+  const [currentUserTask, task] = useMemo(getCurrentUserTask, [product, currentUser]);
```

Comment:
can't use hooks after a return (as on 116 ash 126)

Reply:
react's hook implementation requires that all hooks be used every time the component is rendered. so sometimes having an additional hook (or fewer hooks) throws off the hook manager